test/fetch.test.js

```diff
@@ -0,0 +1,175 @@
+                    authorization: 'Basic 123'
+                }
+            })
+                .post('/path', 'name=value')
+                .reply(200, JSON.stringify(expectedResponse), {
+                    'content-type': 'application/json'
+                });
+    
+            const form = new URLSearchParams();
+            form.append("name", "value");
+    
+            const cookieJar = new tough.CookieJar();
+            const result = await postForm("http://post-form-test/path", form, cookieJar, {
+                headers: {
+                    authorization: 'Basic 123'
+                }
+            });
+            assert.deepStrictEqual(result, expectedResponse);
+        });
+        it('cookie-test', async function() {
+            const cookieJar = new tough.CookieJar();
+            const form = new URLSearchParams();
+            form.append("name", "value");
+
+            // invoke first post, make sure the cookie is set
+            nock('http://post-form-test')
+                .post('/path', 'name=value')
+                .reply(200, JSON.stringify({}), {
+                    'content-type': 'application/json',
+                    'set-cookie': 'id=a3fWa; Max-Age=2592000'
+                });
+            const result1 = await postForm("http://post-form-test/path", form, cookieJar);
+            assert.deepStrictEqual(result1, {});
+            for (const cookie of await cookieJar.getCookies('http://post-form-test')) {
+                assert.strictEqual(cookie.domain, 'post-form-test');
+                assert.strictEqual(cookie.key, 'id');
+                assert.strictEqual(cookie.value, 'a3fWa');
+            }
+
+            // invoke second post, make sure the cookie is returned to the server
+            nock('http://post-form-test', {
+                reqheaders: {
+                    cookie: 'id=a3fWa'
+                }
+            })
+                .post('/path', 'name=value')
+                .reply(200, JSON.stringify({}), {
+                    'content-type': 'application/json'
+                });
+            const result2 = await postForm("http://post-form-test/path", form, cookieJar);
+            assert.deepStrictEqual(result2, {});
+        });
+        it('cookie-auth-test', async function() {
+            // make sure the authorization header is passed through together with the cookie
+            const cookieJar = new tough.CookieJar();
+            const form = new URLSearchParams();
+            form.append("name", "value");
+
+            // invoke first post, make sure the cookie is set
+            nock('http://post-form-test', {
+                reqheaders: {
+                    authorization: 'Basic 123',
+                }
+            })
+                .post('/path', 'name=value')
+                .reply(200, JSON.stringify({}), {
+                    'content-type': 'application/json',
+                    'set-cookie': 'id=a3fWa; Max-Age=2592000'
```

Comment:
Do you need to test also max age handling? What if a cookie becomes invalid during transfer? Is it even possible?